pkg/controllers/proxy/network_services_controller.go

```diff
@@ -1085,6 +1092,223 @@ func (ln *linuxNetworking) prepareEndpointForDsr(containerID string, endpointIP
 	return nil
 }
 
+// The same as prepareEndpointForDsr but using CRI instead of docker
```

Comment:
We should add a comment to `prepareEndpointForDSR`  with this PR saying that it is deprecated and to evaluate removing the previous functionality when Kubernetes 1.23 becomes a prominent release.

Reply:
done